text/0036-async-testbench-functions.md

```diff
@@ -0,0 +1,262 @@
+    Otherwise, a plain integer is returned.
+- `set(expr: Value, value: ConstLike)`
+- `set(expr: ValueCastable, value: any)`
+  - Set `expr` to `value` when awaited.
+    When `expr` is a value-castable, and its `shape()` is a `ShapeCastable`, the value will be converted through the shape's `.const()`.
+    Otherwise, it must be a const-castable `ValueLike`.
+- `memory_read(instance: MemoryIdentity, address: int)`
+  - Read the value from `address` in `instance` when awaited.
+- `memory_write(instance: MemoryIdentity, address: int, value: int, mask:int = None)`
+  - Write `value` to `address` in `instance` when awaited. If `mask` is given, only the corresponding bits are written.
+    Like `MemoryInstance`, these two functions are an internal interface that will be usually only used via `lib.Memory`.
+    It comes without a stability guarantee.
+- `tick(domain="sync", *, context=None)`
+  - Create a domain trigger object for advancing simulation until the next active edge of the `domain` clock.
+    When an elaboratable is passed to `context`, `domain` will be resolved from its perspective.
+  - If `domain` is asynchronously reset while this is being awaited, `amaranth.sim.AsyncReset` is raised.
+- `delay(interval: float)`
+  - Create a combinable trigger object for advancing simulation by `interval` seconds.
+- `changed(*signals)`
+  - Create a combinable trigger object for advancing simulation until any signal in `signals` changes.
+- `edge(signal, value)`
+  - Create a combinable trigger object for advancing simulation until `signal` is changed to `value`.
+    `signal` must be a 1-bit signal or a 1-bit slice of a signal.
+    `value` is a boolean where true indicates rising edge and false indicates falling edge.
+- `posedge(signal)`
+- `negedge(signal)`
+  - Aliases for `edge(signal, 1)` and `edge(signal, 0)` respectively.
+- `critical()`
+  - Context manager.
+    If the current process is a background process, `async with sim.critical():` makes it a non-background process for the duration of the statement.
+
+The `ProcessType` enum has the following members:
+  - `BikeshedProcess`
+    - The current process is a process added by `add_process`.
+  - `BikeshedTestbench`
+    - The current process is a testbench process added by `add_testbench`.
+
+A domain trigger object has the following methods:
+- `__await__()`
+  - Advance simulation. No value is returned.
+- `until(condition)`
+  - Repeat the trigger until `condition` is true.
+    `condition` is an arbitrary Amaranth expression.
+    If `condition` is initially true, `await` will return immediately without advancing simulation.
+    The return value is an unspecified awaitable with `await` as the only defined operation.
+- `repeat(times: int)`
+  - Repeat the trigger `times` times.
+    The return value is an unspecified awaitable with `await` as the only defined operation.
```

Comment:
Is `repeat(0)` allowed? (I.e. is it `times > 0` or `times >= 0`?)

What does awaiting this unspecified awaitable return?